libs/vkd3d/swapchain.c

```diff
@@ -2552,16 +2539,34 @@ static void dxgi_vk_swap_chain_present_callback(void *chain_)
 #endif
 }
 
+static void dxgi_vk_swap_chain_update_frame_statistics(struct dxgi_vk_swap_chain *chain, uint64_t present_count)
+{
+    uint64_t time;
+
+#ifdef _WIN32
+    /* DXGI returns the raw QPC value */
+    LARGE_INTEGER li;
+    QueryPerformanceCounter(&li);
+    time = li.QuadPart;
+#else
+    time = vkd3d_get_current_time_ns();
+#endif
+
+    pthread_mutex_lock(&chain->frame_statistics.lock);
```

Comment:
This seems like a good candidate for a spin-lock, but I don't have a strong opinion.

Reply:
Sounds reasonable.